File: regexlib/javascript_test_file/regexlib_8182.js
/* 8182
 * <\?xml.*</note>
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"<?xml"*5000+"! _1SLQ_2"
 */
var REGEX = new RegExp("<\?xml.*</note>")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '<?xml'.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_8182.py
# 8182
# <\?xml.*</note>
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"<?xml"*5000+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """<\?xml.*</note>"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "<?xml" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")